org.eclipse.scanning.points/src/org/eclipse/scanning/points/CompoundSpgIterator.java

```diff
@@ -106,6 +109,8 @@ public boolean hasNext() {
 //			
 //			if (gen.containsPoint(x, y)) {
 			currentPoint = point;
+			index++;
+			if (currentPoint!=null) currentPoint.setStepIndex(index);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') At most one statement is allowed per line, but 2 statements were found on this line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00122)